### PR TITLE
Add new features in the eris starter

### PR DESCRIPTION
**Pull request by agam778**
- Add a command handler, which makes easy to create new cmds
- Add a help command which scans all the commands in the cmd handler directory
image
- Add a bot status feature
- Add a custom prefix feature (defaults to "!", you can set a custom prefix using adding an env var `PREFIX` with your prefix)
- Add an error handler, so the app doesn't crash if an error occurs
- Update dependencies in Package.json

### CODE REVIEW

**Comment by devtomio** [comment]
Isn't this quite advanced compared to the other bot starters?

File: examples/eris/src/commands/help.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+const fs = require("fs");
```
Comment (devtomio) [inline examples/eris/src/commands/help.js:1]:
```suggestion
const fs = require("node:fs");
```

https://nodejs.org/api/modules.html#core-modules

File: examples/eris/src/index.js [hunk 2]
```diff
@@ -1,14 +1,32 @@
...
 const bot = Eris(`Bot ${process.env.DISCORD_TOKEN}`);
+const prefix = process.env.PREFIX || "!";
+const commands = fs
+  .readdirSync(__dirname + "/commands")
+  .filter(file => file.endsWith(".js"));
 
-bot.on('ready', () => console.log(`Logged in as ${bot.user.username}#${bot.user.discriminator}`));
+bot.on("ready", () =>
+  console.log(`Logged in as ${bot.user.username}#${bot.user.discriminator}`),
+);
```
Comment (devtomio) [inline examples/eris/src/index.js:11]:
```suggestion
bot.on("ready", () => console.log(`Logged in as ${bot.user.username}#${bot.user.discriminator}`));
```

why indent?

File: examples/eris/src/index.js [hunk 3]
```diff
@@ -1,14 +1,32 @@
-const Eris = require('eris');
+const Eris = require("eris");
+const fs = require("fs");
```
Comment (devtomio) [inline examples/eris/src/index.js:2]:
```suggestion
const fs = require("node:fs");
```

https://nodejs.org/api/modules.html#core-modules

File: examples/eris/src/index.js [hunk 4]
```diff
@@ -1,14 +1,32 @@
...
+bot.editStatus("online", {
+  name: "Listening to Your commands <3",
+  type: 0,
 });
 
 bot.connect();
+
+process.on("unhandledRejection" || "uncaughtException", err => {
+  console.error(err);
+});
```
Comment (devtomio) [inline examples/eris/src/index.js:32]:
```suggestion
process.on("unhandledRejection" || "uncaughtException", console.error);
```

File: examples/eris/src/index.js [hunk 5]
```diff
@@ -1,14 +1,32 @@
...
-		bot.createMessage(message.channel.id, 'Pong!');
-	} else if (message.content === 'Hello') {
-		bot.createMessage(message.channel.id, 'Choo choo! 🚅');
-	}
+bot.on("messageCreate", async message => {
+  if (!message.content.startsWith(prefix)) return;
+  if (message.author.bot) return;
+  const args = message.content.slice(prefix.length).split(/ +/);
+  const command = args.shift().toLowerCase();
+  if (!commands.includes(command + ".js")) return;
```
Comment (devtomio) [inline examples/eris/src/index.js:18]:
```suggestion
  if (!commands.includes(`${command}.js`)) return;
```

Use [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals).

File: examples/eris/src/commands/ping.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+module.exports = {
+  name: "ping",
+  description: "Pings the bot",
+  async execute(message, args) {
+    const msg = await message.channel.createMessage("Pinging...");
+    const ping = msg.timestamp - message.timestamp;
+    msg.edit("🏓 Pong! `" + ping + "ms`");
```
Comment (devtomio) [inline examples/eris/src/commands/ping.js:7]:
```suggestion
    msg.edit(`🏓 Pong! \`${ping} ms\``);
```

Use [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals).

File: examples/eris/src/commands/hello.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
+module.exports = {
+  name: "hello",
+  description: "Says hello!",
+  async execute(message, args) {
+    message.channel.createMessage(
+      "Hello " + message.author.username + "!" + " Choo choo 🚅",
+    );
```
Comment (devtomio) [inline examples/eris/src/commands/hello.js:7]:
```suggestion
    message.channel.createMessage(`Hello ${message.author.username}!, Choo choo 🚅`);
```

Dont indent and use [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals).

File: examples/eris/src/commands/hello.js [hunk 8]
```diff
@@ -0,0 +1,9 @@
+module.exports = {
+  name: "hello",
+  description: "Says hello!",
+  async execute(message, args) {
```
Comment (devtomio) [inline examples/eris/src/commands/hello.js:4]:
```suggestion
  async execute(message) {
```

Variable not used.

File: examples/eris/src/commands/help.js [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+module.exports = {
+  name: "help",
+  description: "Shows help for all commands",
+  async execute(message, args) {
```
Comment (devtomio) [inline examples/eris/src/commands/help.js:5]:
```suggestion
  async execute(message) {
```

Variable not used.

File: examples/eris/src/commands/ping.js [hunk 10]
```diff
@@ -0,0 +1,9 @@
+module.exports = {
+  name: "ping",
+  description: "Pings the bot",
+  async execute(message, args) {
```
Comment (devtomio) [inline examples/eris/src/commands/ping.js:4]:
```suggestion
  async execute(message) {
```

Variable not used.

File: examples/eris/src/index.js [hunk 11]
```diff
@@ -1,14 +1,32 @@
-const Eris = require('eris');
+const Eris = require("eris");
+const fs = require("fs");
 const bot = Eris(`Bot ${process.env.DISCORD_TOKEN}`);
+const prefix = process.env.PREFIX || "!";
```
Comment (devtomio) [inline examples/eris/src/index.js:4]:
```suggestion
const prefix = process.env.PREFIX;
```

No need, already defaults to `!` when using the starter.

**Review by devtomio — CHANGES_REQUESTED** [review]

File: examples/eris/src/index.js (hunk 2 above)
Comment (agam778) [reply examples/eris/src/index.js:11]: Prettier did that lol

File: examples/eris/src/index.js (hunk 11 above)
Comment (agam778) [reply examples/eris/src/index.js:4]: No, but like if user wants his own prefix while using the starter without editing the code, this is a good thing then

**Comment by agam778** [comment]
> Isn't this quite advanced compared to the other bot starters?

It's still pretty simple, lol

**Comment by Milo123459** [comment]
This should probably use slash commands.

**Comment by agam778** [comment]
> This should probably use slash commands.

Does the stable release of Eris even have slash commands?

**Comment by Milo123459** [comment]
No, because it's abandoned. This really shouldn't even be a starter imo.

**Comment by agam778** [comment]
Huh? It isn't abondened?

**Comment by Milo123459** [comment]
If the latest stable release doesn't have a feature that is going to become required by the end of April, well...

**Comment by devtomio** [comment]
> If the latest stable release doesn't have a feature that is going to become required by the end of April, well...

Its still being used, and the maintainer will update it when its really needed. Plus, their discord server is quite active so it being "abandoned" is quite contradictory.

**Comment by FarazPatankar** [comment]
Hey @agam778, I am closing this PR as we've restructured how we manage templates. Do you mind opening an issue in the [new repo](https://github.com/railwayapp-templates/discord-eris) discussing these changes before opening a PR implementing them?

**Pull request closed without merging**